src/platform/unix/mod.rs

```diff
@@ -471,8 +471,18 @@ impl OsIpcReceiverSet {
     pub fn select(&mut self) -> Result<Vec<OsIpcSelectionResult>,UnixError> {
         let mut selection_results = Vec::new();
         match self.poll.poll(&mut self.events, None) {
-            Ok(sz) if sz > 0 => {},
-            _ => { return Err(UnixError::last()); }
+            Ok(sz) => {
+                if sz < 1 {
+                    return self.select();
```

Comment:
In testing `poll` always returned `0` after each call that returned `EINTR`. I need to do some more testing and debugging around this.